### PR TITLE
[AIC-1911] Make Fast and Full CI fail closed

### PR DESCRIPTION
**Pull request by jasonqinzhou**
## Why and what changed

Fast and Full CI currently expose conditional jobs, but GitHub can treat a conditionally skipped job as successful. This change adds stable fail-closed results for the exact pull-request head, rejects stale trusted copies, and keeps the work limited to CI enforceability.

## Review map

- Risk level: medium
- Start with: `.github/workflows/fast-ci.yml`, then `.github/workflows/ci.yml`, then the executable contracts in `.github/codeowners/test_aisimulate_policy.py`
- Public or serialized contract changed: no
- Compatibility or rollback concern: branch rules must require the direct `Fast CI Success` and aggregate `Full CI Success` checks only after this workflow behavior is approved. Rollback is removal of the readiness jobs before ruleset activation.

## Evidence

- Tests: `uv run --no-project --with pytest==8.4.2 --with pyyaml==6.0.3 python -m pytest -c /dev/null .github/codeowners/test_aisimulate_policy.py -q -p no:cacheprovider --override-ini=addopts= --override-ini=filterwarnings=` — 10 passed
- Ruff: `uv run --no-project --with ruff==0.14.1 ruff check --config python/aisimulate/pyproject.toml .github/codeowners/test_aisimulate_policy.py` — passed
- Workflow syntax: actionlint v1.7.7 on both modified workflows — passed
- Source and legal: `python3 scripts/check_copyright.py` and `python3 scripts/check_packaged_legal_files.py` — passed for 1,064 tracked source files and packaged notices
- Fast CI: exact head `26d60fbd164cc272ba5337fcd0b558bba8bce8e7` passed in [run 34277235590](https://github.com/ai-dynamo/aisimulate/actions/runs/34277235590)
- Full CI: exact head `26d60fbd164cc272ba5337fcd0b558bba8bce8e7` passed on the trusted `pull-request/135` copy in [run 34279933019](https://github.com/ai-dynamo/aisimulate/actions/runs/34279933019); `Full CI Success` passed and release staging was explicitly N/A/skipped for PR validation
- CodeRabbit reviewed commit: `26d60fbd164cc272ba5337fcd0b558bba8bce8e7`; all three actionable threads were fixed and automatically resolved
- Codex reviewed commit: `26d60fbd164cc272ba5337fcd0b558bba8bce8e7`; no remaining actionable code finding
- Negative or boundary cases: executable tests cover missing `review-ready`, skipped/canceled/missing dependencies, empty and mismatched manual SHA, matching/stale/invalid/API-failed trusted copies, draft behavior, and PR-versus-release staging. A controlled non-draft run without `review-ready` failed as intended in [run 34276061915](https://github.com/ai-dynamo/aisimulate/actions/runs/34276061915), followed by a same-head passing labeled run.
- Expected-value derivation: job results must equal GitHub Actions terminal `success`; release staging must be `success` for `main` and `release/*` pushes and `skipped` for manual or trusted-copy PR validation
- Before/after output: before, skipped conditional jobs could satisfy a required check; after, `Fast CI Success` failed on the unlabeled controlled run and passed only after applicable Fast jobs succeeded

## Modeling or data provenance

N/A. No simulator formula, model data, selection logic, serialized field, prediction, or golden changed.

## Tracking

- Related: [AIC-1911 — Make Fast and Full CI unavoidable exact-head gates](https://linear.app/nvidia/issue/AIC-1911/ci-make-fast-and-full-ci-unavoidable-exact-head-gates)
- Related test-bucketing work: [AIC-1931](https://linear.app/nvidia/issue/AIC-1931/ci-rebucket-pr-96-validation-gates-into-fast-ci-and-full-ci)
- Existing Fast/Full foundation: [PR #118](https://github.com/ai-dynamo/aisimulate/pull/118)

### CODE REVIEW

File: .github/codeowners/test_aisimulate_policy.py [hunk 1]
```diff
@@ -216,6 +239,66 @@ def test_fast_and_full_ci_keep_their_cost_boundary() -> None:
...
+        "application-tests",
+        "release-artifact-contract",
+        "application-wheel",
+        "stage-application-wheel",
+    }
+    full_readiness_script = full_readiness["steps"][0]["run"]
+    assert "${VERIFY_TARGET_RESULT}" in full_readiness_script
+    assert "${APPLICATION_WHEEL_RESULT}" in full_readiness_script
+    assert "${STAGE_APPLICATION_WHEEL_RESULT}" in full_readiness_script
+    assert '"${STAGE_APPLICATION_WHEEL_RESULT}" "skipped"' in full_readiness_script
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/codeowners/test_aisimulate_policy.py:309]: Fixed in `26d60fb`: the policy test now normalizes and compares the staging job condition with the `STAGING_REQUIRED` readiness expression, so either side drifting fails the contract.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -424,3 +449,79 @@ jobs:
...
+    needs:
+      - verify-target
+      - fast-ci
+      - cargo-deny
+      - rust
+      - public-api-rust
+      - application-tests
+      - release-artifact-contract
+      - application-wheel
+      - stage-application-wheel
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/ci.yml:465]: Fixed in `26d60fb`: the test now asserts that the Full CI readiness job depends on every other workflow job, in addition to retaining the explicit expected-job set.

File: REVIEW.md [hunk 3]
```diff
@@ -47,6 +47,15 @@ finding. Lower-priority findings and CODEOWNER review may proceed while Full CI
 runs, but all required conversations, approvals, and exact-head checks must be
 complete before merge.
 
+`Fast CI Success` and `Full CI Success` are the stable merge-gate results. Both
+run with `always()` semantics and fail when required evidence is missing,
+skipped unexpectedly, canceled, or failed. A non-draft PR without the
+`review-ready` label fails `Fast CI Success`; making a PR ready or removing the
+label retriggers the workflow. Release staging is explicitly not applicable to
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply REVIEW.md]: Fixed in `26d60fb`: the policy test now pins the required pull-request activity types, and the workflow comments plus `REVIEW.md` document the direct and reusable check names. Branch rules require direct `Fast CI Success` and aggregate `Full CI Success`; the composed reusable check is covered by the latter.

**Review by tedzhouhk — APPROVED** [review]
